android/src/main/kotlin/com/airship/flutter/FlutterAutopilot.kt

```diff
@@ -22,29 +19,68 @@ class FlutterAutopilot : Autopilot() {
-                    EventManager.shared.notifyEvent(DeepLinkEvent(deepLink))
-                }
-                return ActionResult.newResult(arguments.value)
+        airship.pushManager.addPushListener{ message, notificationPosted ->
+            if (!notificationPosted) {
+                post(PushReceivedEvent(message))
             }
         }
 
-        // Inbox
-        airship.actionRegistry.getEntry(OpenRichPushInboxAction.DEFAULT_REGISTRY_NAME).defaultAction = object : OpenRichPushInboxAction() {
-            override fun perform(arguments: ActionArguments): ActionResult {
-                val messageId = arguments.value.string
+        airship.pushManager.setNotificationListener(object : NotificationListener {
+
+            override fun onNotificationPosted(@NonNull notificationInfo: NotificationInfo) {
+                post(PushReceivedEvent(notificationInfo.message, notificationInfo))
+            }
+
+            override fun onNotificationOpened(@NonNull notificationInfo: NotificationInfo): Boolean {
+                post(NotificationResponseEvent(notificationInfo))
+
+                return false
+            }
+
+            override fun onNotificationForegroundAction(@NonNull notificationInfo: NotificationInfo, @NonNull notificationActionButtonInfo: NotificationActionButtonInfo): Boolean {
+                post(NotificationResponseEvent(notificationInfo, notificationActionButtonInfo))
+
+                return false
+            }
+
+            override fun onNotificationBackgroundAction(@NonNull notificationInfo: NotificationInfo, @NonNull notificationActionButtonInfo: NotificationActionButtonInfo) {
+                post(NotificationResponseEvent(notificationInfo, notificationActionButtonInfo))
+
+            }
+
+            override fun onNotificationDismissed(@NonNull notificationInfo: NotificationInfo) {}
+        })
+
+        airship.pushManager.addRegistrationListener(object : RegistrationListener {
+            override fun onChannelCreated(channelId: String) {
+                post(ChannelRegistrationEvent(channelId, UAirship.shared().pushManager.registrationToken))
+            }
+
+            override fun onChannelUpdated(channelId: String) {
+                post(ChannelRegistrationEvent(channelId, UAirship.shared().pushManager.registrationToken))
+            }
+
+            override fun onPushTokenUpdated(pushToken: String) {}
+        })
+
+        airship.messageCenter.setOnShowMessageCenterListener(object : MessageCenter.OnShowMessageCenterListener {
+            override fun onShowMessageCenter(messageId: String?): Boolean {
                 if (messageId != null) {
                     EventManager.shared.notifyEvent(ShowInboxMessageEvent(messageId))
-                } else {
-                    EventManager.shared.notifyEvent(ShowInboxEvent())
                 }
 
-                return ActionResult.newEmptyResult()
+                EventManager.shared.notifyEvent(ShowInboxEvent())
```

Comment:
Just to verify: both events when you get a messageId?

Reply:
needed an early return